plugins/network-elements/nuage-vsp/src/com/cloud/agent/api/element/ImplementVspCommand.java

```diff
@@ -0,0 +1,295 @@
+        if (_isL2Network != that._isL2Network) return false;
+        if (_isL3Network != that._isL3Network) return false;
+        if (_isShared != that._isShared) return false;
+        if (_isVpc != that._isVpc) return false;
+        if (_networkId != that._networkId) return false;
+        if (_acsFipUuid != null ? !_acsFipUuid.equals(that._acsFipUuid) : that._acsFipUuid != null) return false;
+        if (_dnsServers != null ? !_dnsServers.equals(that._dnsServers) : that._dnsServers != null) return false;
+        if (_domainTemplateName != null ? !_domainTemplateName.equals(that._domainTemplateName) : that._domainTemplateName != null)
+            return false;
+        if (_egressFirewallRules != null ? !_egressFirewallRules.equals(that._egressFirewallRules) : that._egressFirewallRules != null)
+            return false;
+        if (_ingressFirewallRules != null ? !_ingressFirewallRules.equals(that._ingressFirewallRules) : that._ingressFirewallRules != null)
+            return false;
+        if (_networkDomainUuid != null ? !_networkDomainUuid.equals(that._networkDomainUuid) : that._networkDomainUuid != null)
+            return false;
+        if (_networkName != null ? !_networkName.equals(that._networkName) : that._networkName != null) return false;
+        if (_networkUuid != null ? !_networkUuid.equals(that._networkUuid) : that._networkUuid != null) return false;
+        if (_vpcOrSubnetUuid != null ? !_vpcOrSubnetUuid.equals(that._vpcOrSubnetUuid) : that._vpcOrSubnetUuid != null)
+            return false;
+
+        return true;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = (int) (_networkId ^ (_networkId >>> 32));
+        result = 31 * result + (_networkDomainUuid != null ? _networkDomainUuid.hashCode() : 0);
+        result = 31 * result + (_networkUuid != null ? _networkUuid.hashCode() : 0);
+        result = 31 * result + (_networkName != null ? _networkName.hashCode() : 0);
+        result = 31 * result + (_vpcOrSubnetUuid != null ? _vpcOrSubnetUuid.hashCode() : 0);
+        result = 31 * result + (_isL2Network ? 1 : 0);
+        result = 31 * result + (_isL3Network ? 1 : 0);
+        result = 31 * result + (_isVpc ? 1 : 0);
+        result = 31 * result + (_isShared ? 1 : 0);
+        result = 31 * result + (_domainTemplateName != null ? _domainTemplateName.hashCode() : 0);
+        result = 31 * result + (_isFirewallServiceSupported ? 1 : 0);
+        result = 31 * result + (_dnsServers != null ? _dnsServers.hashCode() : 0);
+        result = 31 * result + (_ingressFirewallRules != null ? _ingressFirewallRules.hashCode() : 0);
+        result = 31 * result + (_egressFirewallRules != null ? _egressFirewallRules.hashCode() : 0);
+        result = 31 * result + (_acsFipUuid != null ? _acsFipUuid.hashCode() : 0);
+        result = 31 * result + (_egressDefaultPolicy ? 1 : 0);
+        return result;
+    }
```

Comment:
Please implement ``toString()`` to provide debugging state information for debugging purposes.